src/MainScreen.tsx

```diff
@@ -1,88 +1,99 @@
-import React, { useState } from 'react';
-import { View, Text, StyleSheet, TextInput, Button, Alert, SafeAreaView } from 'react-native';
+import React, { useState, useCallback } from 'react';
+import { View, Text, StyleSheet, TextInput, Button, SafeAreaView, Alert } from 'react-native';
 
 import { CarData } from './types/CarData';
 
 type MainScreenProps = {
-  onPressSave?: (carData: CarData) => void;
-  onPressGet?: (carGetDeleteId: string) => void;
-  onPressDelete?: (carGetDeleteId: string) => void;
-};
-
-const storageErrorFunc = () => {
-  Alert.alert('Storage access error');
+  saveInStorage: (carId: string, carData: CarData) => void;
+  getFromStorage: (carId: string) => void;
+  deleteFromStorage: (carId: string) => void;
 };
 
 const MainScreen: React.FC<MainScreenProps> = ({
-  onPressSave = storageErrorFunc,
-  onPressGet = storageErrorFunc,
-  onPressDelete = storageErrorFunc,
+  saveInStorage,
+  getFromStorage,
+  deleteFromStorage,
 }) => {
-  const [carData, setCarData] = useState({ carId: '', year: '', make: '', model: '' });
-  const [carGetId, setCarGetId] = useState('');
+  const [carSaveData, setCarSaveData] = useState({
+    carId: '',
+    year: '',
+    make: '',
+    model: '',
+  });
+  const [carId, setCarId] = useState('');
+
+  const onPressSave = useCallback(() => {
+    if (!carSaveData.carId || !carSaveData.year || !carSaveData.make || !carSaveData.model) {
+      Alert.alert('Some field is empty', 'Please write data to all fileds');
```

Comment:
typo `fileds`. It's better to say: `Unable to save car details. Please, fill all required fields`